examples/manual_tests/respawn_entity.rs

```diff
@@ -0,0 +1,80 @@
+use std::time::Duration;
+
+use bevy::{app::ScheduleRunnerPlugin, log::LogPlugin, prelude::*, utils::HashMap};
+use bevy_ggrs::{
+    ggrs::{Config, PlayerType, SessionBuilder},
+    prelude::*,
+    LocalInputs, LocalPlayers,
+};
+
+#[derive(Debug)]
+pub struct GgrsConfig;
+impl Config for GgrsConfig {
+    type Input = u8;
+    type State = u8;
+    type Address = String;
+}
+
+pub fn read_local_input(mut commands: Commands, local_players: Res<LocalPlayers>) {
+    dbg!(&local_players.0);
+    let mut local_inputs = HashMap::new();
+    for handle in &local_players.0 {
+        local_inputs.insert(*handle, 0);
+    }
+    commands.insert_resource(LocalInputs::<GgrsConfig>(local_inputs));
+}
+
+/// Player health
+#[derive(Component, Hash, Debug, Clone, Copy)]
+pub struct Health(u32);
+
+impl Default for Health {
+    fn default() -> Self {
+        Self(3)
+    }
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let mut session = SessionBuilder::<GgrsConfig>::new()
+        .with_num_players(1)
+        // each frame, roll back and resimulate 5 frames back in time, and compare checksums
+        .with_check_distance(2);
```

Comment:
Won't a `with_check_distance` of 2 only resimulate 5 frames?

Reply:
It's better to increase it yes. I set it to to for a very minimal repro of the issue, but should revert that last commit.

Reply:
I just realized I mistyped my comment. I meant to say: "Won't a `with_check_distance` of 2 only resimulate 2 frames?" The comment says 5.